plugins/net_plugin/net_plugin.cpp

```diff
@@ -3464,15 +3823,31 @@ namespace eosio {
          }
          ++it;
       }
+      if(it == connections.end())
```

Comment:
`it == connections.end()` seems guaranteed at this point no? I see no breaks that would short circuit the main while condition